tests/testthat/test-stan_discretised_logit_hazard.R

```diff
@@ -69,3 +69,93 @@ test_that("discretised_logit_hazard returns the same thing in both log
   expect_equal_prob(2, 1, 4)
   expect_equal_prob(3, 0.2, 4)
 })
+
+# Discretisation for double censoring
+double_censored_pmf <- function(n, alpha, beta, fun = plnorm) {
+  cdf <- fun(1:(n+1), alpha, beta)
+  pmf  <- vector("numeric", n)
+  pmf[1] <- cdf[1]
+  pmf[2] <- cdf[2] 
+  pmf[3:n] <- cdf[3:n] - cdf[1:(n-2)]
+  pmf <- pmf / sum(pmf)
+  return(pmf)
+}
+
+# Discretisation for single censoring
+single_censored_pmf <- function(n ,alpha, beta, fun = plnorm) {
+  cdf <- fun(1:(n+1), alpha, beta)
+  pmf  <- vector("numeric", n)
+  pmf[1] <- cdf[1]
+  pmf[2:n] <- cdf[2:n] - cdf[1:(n-1)]
+  pmf <- pmf / cdf[n]
+  return(pmf)
+}
+
+# Simulate double censored data
+simulate_double_censored_pmf <- function(
+  alpha, beta, max, fun = rlnorm, n = 1000
+) {
+  primary <- runif(n, 0, 1)
+  secondary <- primary + fun(n, alpha, beta)
```

Comment:
Can you just help me here, why you have done it like this? I would have intuitively also expected a runif for `secondary`

Reply:
Me as well which it turns out is wrong! 

So this is saying that `rprimary` can happen anywhere within a day. The secondary event then depends on this time and a draw from the probability distribution. 

The reason that the secondary event doesn't need a uniform is that this is captured in the observation process (the flooring). We then observe the delay which is the floored to a day version of the secondary event - the floored to a day version of primary event. 

Attached is a figure I have been playing around with to get at this. It shows the growth rate based real process compared two various approximations. The one that is equivalent to what we have here is the Uniform(-1, 1) one. 
[censoring-intervals.pdf](https://github.com/epinowcast/epinowcast/files/11789290/censoring-intervals.pdf)

Reply:
We really want the version that has a uniform on both primary and secondary but I don't have a good handle on what that looks like yet.

Reply:
Does this make sense now @adrian-lison?

Reply:
Happy with this as is?